catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -224,22 +219,30 @@ private Object configure() {
   @SuppressWarnings(
       "squid:S1172" /* The code parameter is required in blueprint-cm-1.0.7. See https://issues.apache.org/jira/browse/ARIES-1436. */)
   public void destroy(int code) {
-    if (routeCollection == null) {
-      return;
-    }
+    synchronized (lock) {
```

Comment:
❗️ ☝️ There is no point using `CompletableFuture.runAsync()` if the returned future isn't used. Simply go directly to the `configurationExecutor` to execute `this::attemptAddRoutes`